src/CoreBundle/private/library/classes/dbobjects/TCMSPage.class.php

```diff
@@ -446,6 +433,50 @@ public function GetRealURL($aAdditionalParameters = array(), $aExcludeParameters
         return $sLink;
     }
 
+    /**
+     * returns all POST and GET parameters as url.
+     *
+     * @param array $excludeArray
+     *
+     * @return string
+     */
+    private function OutputDataAsURL($excludeArray = array())
+    {
+        $aData = TGlobal::instance()->GetUserData();
```

Comment:
This is also deprecated (?)

Reply:
The method was just moved from TGlobalBase without changes. A handful of extra usages won't hurt and can be changed later (touching hundreds of files, so I'd like to avoid effort and errors for now).

Reply:
You want to revisit this with another PR (because I think the issue will be the same)?

Reply:
We will keep some deprecated code in 7.0 as it's too much to remove everything at once. TGlobalBase::Instance() is definitely one of those methods that will stay, as there are currently 327 usages in chameleon-base and chameleon-shop alone. We will need to reduce usage of those methods in an ongoing effort so that it can then be removed in 8.0.

The ultimate goal with TGlobal and TGlobalBase (and TTools) should be to remove them completely as they are global util classes that should never have existed. So there will be more deprecations in future minor releases and less and less need for the instance() method, so some of the work should be done "automatically".